src/main/java/io/quarkus/bot/PingWhenNeedsTriageRemoved.java

```diff
@@ -0,0 +1,125 @@
+package io.quarkus.bot;
+
+import io.quarkiverse.githubapp.ConfigFile;
+import io.quarkiverse.githubapp.event.Issue;
+import io.quarkus.bot.config.QuarkusBotConfig;
+import io.quarkus.bot.config.QuarkusBotConfigFile;
+import io.quarkus.bot.el.SimpleELContext;
+import io.quarkus.bot.util.Labels;
+import io.quarkus.bot.util.Patterns;
+import io.quarkus.bot.util.Strings;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHLabel;
+
+import javax.el.ELContext;
+import javax.el.ELManager;
+import javax.el.ExpressionFactory;
+import javax.el.ValueExpression;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class PingWhenNeedsTriageRemoved {
+    private static final Logger LOG = Logger.getLogger(PingWhenNeedsTriageRemoved.class);
+
+    @Inject
+    QuarkusBotConfig quarkusBotConfig;
+
+    void pingWhenNeedsTriageRemoved(@Issue.Unlabeled GHEventPayload.Issue issuePayload,
+            @ConfigFile("quarkus-bot.yml") QuarkusBotConfigFile quarkusBotConfigFile) throws IOException {
+
+        if (quarkusBotConfigFile == null) {
+            LOG.error("Unable to find triage configuration.");
+            return;
+        }
+
+        GHIssue issue = issuePayload.getIssue();
+        GHLabel removedLabel = issuePayload.getLabel();
+        Set<String> mentions = new TreeSet<>();
+
+        for (QuarkusBotConfigFile.TriageRule rule : quarkusBotConfigFile.triage.rules) {
+            if (matchRule(issue, rule)) {
+                if (!rule.notify.isEmpty()) {
+                    for (String mention : rule.notify) {
+                        if (!mention.equals(issue.getUser().getLogin())) {
+                            mentions.add(mention);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (removedLabel.getName().equals(Labels.TRIAGE_NEEDS_TRIAGE)) {
+            if (!mentions.isEmpty()) {
+                if (!quarkusBotConfig.isDryRun()) {
+                    issue.comment("/cc @" + String.join(", @", mentions));
+                } else {
+                    LOG.info("Issue #" + issue.getNumber() + " - Ping: " + mentions);
+                }
+            }
+        }
+
+    }
+
+    private static boolean matchRule(GHIssue issue, QuarkusBotConfigFile.TriageRule rule) {
+        try {
+            if (Strings.isNotBlank(rule.title)) {
+                if (Patterns.find(rule.title, issue.getTitle())) {
```

Comment:
@gsmet so here instead of applying rules on title, body, etc I should apply the rule on the labels present in that issue 🤔

Reply:
Exactly, the idea is to match the labels that were potentially added with the ones in the triage config file and then ping the persons.

Think of it this way:
- an issue has not been properly triaged as no rules matched for the content
- I add labels manually to classify the issue
- I remove the `needs-triage` label
- the bot automatically pings the people for the labels I manually added
- nice win because I don't have to figure out who is responsible for these areas